portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -5873,6 +5873,10 @@ public long getUserId(HttpServletRequest httpServletRequest) {
 
 		HttpSession httpSession = httpServletRequest.getSession();
 
+		if (httpSession == null) {
+			return 0;
+		}
+
```

Comment:
@jonathanmccann

What is the case that the httpSession will be null?

Reply:
Same as above.